app/src/androidTest/java/ch/epfl/sdp/musiconnect/RoomDatabaseTest.java

```diff
@@ -0,0 +1,191 @@
+package ch.epfl.sdp.musiconnect;
+
+import androidx.test.rule.ActivityTestRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import ch.epfl.sdp.musiconnect.roomdatabase.AppDatabase;
+import ch.epfl.sdp.musiconnect.roomdatabase.InstrumentConverter;
+import ch.epfl.sdp.musiconnect.roomdatabase.MusicianDao;
+import ch.epfl.sdp.musiconnect.roomdatabase.MyDateConverter;
+import ch.epfl.sdp.musiconnect.roomdatabase.MyLocationConverter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+//@RunWith(AndroidJUnit4.class)
+public class RoomDatabaseTest {
+
+    private AppDatabase roomDb;
+    private MusicianDao musicianDao;
+    private Executor mExecutor = Executors.newSingleThreadExecutor();
+    private List<Musician> allUsers = new ArrayList<>();
+    private List<Musician> users = new ArrayList<>();
+
+
+
+    @Rule
+    public final ActivityTestRule<StartPage> startPageRule =
+            new ActivityTestRule<>(StartPage.class);
+
+    @Before
+    public void saveDatabaseContentsBeforeTesting() {
+        roomDb = AppDatabase.getInstance(startPageRule.getActivity().getApplicationContext());
+        musicianDao = roomDb.musicianDao();
+        mExecutor.execute(() -> {
+            allUsers = musicianDao.getAll();
+            musicianDao.nukeTable();
+        });
+    }
+
```

Comment:
There might be a dialog asking for location permission that can block the screen, I will add a call to click on the alert